### PR TITLE
board: arm: Add PhyBOARD Pollux

### PR DESCRIPTION
**Pull request by pefech**
Add board definition to run Zephyr on the M7-Core
of the i.MX8M Plus SoC on the PhyBOARD Pollux.

All configured interfaces were tested and are working.

Signed-off-by: Peter Fecher <p.fecher@phytec.de>

### CODE REVIEW

File: boards/arm/mimx8mp_phyboard_pollux/mimx8mp_phyboard_pollux_defconfig
Comment (danieldegrasse) [inline boards/arm/mimx8mp_phyboard_pollux/mimx8mp_phyboard_pollux_defconfig:16]: If `CONFIG_XIP=y` here, then why do you provide defaults for `FLASH_BASE_ADDRESS` and `FLASH_SIZE` when it is disabled in the `Kconfig.defconfig` for this board?

**Comment by dleach02** [comment]
@pefech will you be returning to this PR?

**Comment by pefech** [comment]
My priority was on other tasks. I'll continue to work on this PR soon.

**Comment by pefech** [comment]
@danieldegrasse I removed the ```CONFIG_XIP```option from the ```mimx8mp_phyboard_pollux_defconfig``` file. The Board works with XIP enabled and disabled. Now it can be configured in the programs ```prj.conf```file.

**Review by danieldegrasse — APPROVED** [review]

**Comment by pefech** [comment]
Is there any chance someone has some time to make the second review on this?

**Comment by danieldegrasse** [comment]
@JiafeiPan Could you take a look?

File: boards/arm/mimx8mp_phyboard_pollux/doc/index.rst
Comment (pefech) [reply boards/arm/mimx8mp_phyboard_pollux/doc/index.rst:101]: You are right, I'll change that.

File: boards/arm/mimx8mp_phyboard_pollux/mimx8mp_phyboard_pollux-pinctrl.dtsi
Comment (pefech) [reply boards/arm/mimx8mp_phyboard_pollux/mimx8mp_phyboard_pollux-pinctrl.dtsi:17]: I thought it makes sense to have every uart that is possible to use with the M7-Core preconfigured but set on disabled to make it easy to use it in an application. Maybe someone wants to use the uart1 for the M7-Core and another one for the Coretex-A53. Like this it can easily changed, because its preconfigured. Do you think different about that?

File: boards/arm/mimx8mp_phyboard_pollux/doc/index.rst
Comment (pefech) [reply boards/arm/mimx8mp_phyboard_pollux/doc/index.rst:144]: I'll check again, why I wrote this in the table

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
